tools/introspection/source/introspection_app.cpp

```diff
@@ -332,31 +332,31 @@ void IntrospectionApp::printPortIntrospectionData(const std::vector<ComposedPubl
     wprintw(pad, "----------------------------------------------------------------------------\n");
 
     bool needsLineBreak{false};
-    uint32_t currentLine{0};
-    auto printEntry = [&](std::uint32_t maxSize, const std::string& data) -> std::string {
+    uint32_t currentLine{0U};
+    auto printEntry = [&](int32_t maxSize, const std::string& data) -> std::string {
         std::stringstream stream;
 
-        constexpr uint32_t indentation{2};
+        constexpr int32_t indentation{2};
         constexpr char indentationString[indentation + 1] = "  ";
 
         auto stringSize = data.size();
-        if (currentLine == 0)
+        if (currentLine == 0U)
         {
-            stream << std::left << std::setw(maxSize) << data.substr(0, maxSize);
+            stream << std::left << std::setw(maxSize) << data.substr(0U, static_cast<size_t>(maxSize));
         }
-        else if (stringSize > maxSize + (currentLine - 1) * (maxSize - indentation))
+        else if (stringSize > static_cast<size_t>(maxSize) + (currentLine - 1U) * static_cast<size_t>(maxSize - indentation))
         {
-            const auto startPosition = maxSize + (currentLine - 1) * (maxSize - indentation);
+            const auto startPosition = static_cast<size_t>(maxSize) + (currentLine - 1U) * static_cast<size_t>(maxSize - indentation);
 
             stream << indentationString << std::left << std::setw(maxSize - indentation)
-                   << data.substr(startPosition, maxSize - indentation);
+                   << data.substr(startPosition, static_cast<size_t>(maxSize - indentation));
```

Comment:
The casts are getting out of hand and impair readability. Could this have been avoided if e.g. ``maxSize`` and ``indentation`` were already of the correct type (size_t here)? Regardless, it is just a style issue.

Reply:
last time I tried to sanitize this in the introspection app it became wild pretty quick. We need to refactor the whole app. There were a few changes too much since the initial design.